packages/dgrid/src/DatasourceTable.ts

```diff
@@ -24,16 +24,31 @@ export class DatasourceTable extends Common {
 
     update(domNode, element) {
         super.update(domNode, element);
-        if (this._prevDatasource !== this.datasource()) {
-            this._prevDatasource = this.datasource();
-            if (this._prevDatasource) {
+    }
+    render(callback?: (w: Widget) => void): this {
+        return super.render(w => {
+            if (this._prevDatasource !== this.datasource()) {
```

Comment:
This type of logic makes me think this whole concept should belong to common or api